res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,974 @@
+
+    this.linkOutput("[Channel1]", "cue_indicator", this.outputHandler);
+    this.linkOutput("[Channel2]", "cue_indicator", this.outputHandler);
+
+    this.linkOutput("[Channel1]", "sync_enabled", this.outputHandler);
+    this.linkOutput("[Channel2]", "sync_enabled", this.outputHandler);
+
+    this.linkOutput("[Channel1]", "keylock", this.outputHandler);
+    this.linkOutput("[Channel2]", "keylock", this.outputHandler);
+
+    for (var i = 1; i <= 8; ++i) {
+        this.linkOutput("[Channel1]", "hotcue_" + i + "_enabled", this.hotcueOutputHandler);
+        this.linkOutput("[Channel2]", "hotcue_" + i + "_enabled", this.hotcueOutputHandler);
+    }
+
+    this.linkOutput("[Channel1]", "pfl", this.outputHandler);
+    this.linkOutput("[Channel2]", "pfl", this.outputHandler);
+
+    this.linkOutput("[Microphone]", "talkover", this.outputHandler);
+
+    // VuMeter
+    this.vuLeftConnection = engine.makeConnection("[Channel1]", "VuMeter", this.vuMeterHandler);
+    this.vuRightConnection = engine.makeConnection("[Channel2]", "VuMeter", this.vuMeterHandler);
+    this.clipLeftConnection = engine.makeConnection("[Channel1]", "PeakIndicator", this.peakOutputHandler);
+    this.clipRightConnection = engine.makeConnection("[Channel2]", "PeakIndicator", this.peakOutputHandler);
+
+    // Sampler callbacks
+    for (i = 1; i <= 16; ++i) {
+        this.samplerCallbacks.push(engine.makeConnection("[Sampler" + i + "]", "track_loaded", this.samplesOutputHandler));
+        this.samplerCallbacks.push(engine.makeConnection("[Sampler" + i + "]", "play_indicator", this.samplesOutputHandler));
+    }
+
+    TraktorS2MK3.lightDeck(false);
+};
+
+/* Helper function to link output in a short form */
+TraktorS2MK3.linkOutput = function (group, name, callback) {
+    TraktorS2MK3.controller.linkOutput(group, name, group, name, callback);
+};
+
+TraktorS2MK3.vuMeterHandler = function (value, group, key) {
+    var vuKeys = Object.keys(TraktorS2MK3.vuMeterThresholds);
+    for (var i = 0; i < vuKeys.length; ++i) {
+        // Avoid spamming HID by only sending last LED update
+        var last = (i === (vuKeys.length - 1));
+        if (TraktorS2MK3.vuMeterThresholds[vuKeys[i]] > value) {
+            TraktorS2MK3.controller.setOutput(group, vuKeys[i], 0x00, last);
+        } else {
+            TraktorS2MK3.controller.setOutput(group, vuKeys[i], 0x7E, last);
+        }
+    }
+};
+
+TraktorS2MK3.peakOutputHandler = function (value, group, key) {
+    var ledValue = 0x00;
+    if (value) {
+        ledValue = 0x7E;
+    }
+
+    TraktorS2MK3.controller.setOutput(group, key, ledValue, true);
+};
+
+TraktorS2MK3.outputHandler = function (value, group, key) {
+    // Custom value for multi-colored LEDs
+    var ledValue = value;
+    if (value === 0 || value === false) {
+        // Off value
+        ledValue = 0x7C;
+    } else if (value === 1 || value === true) {
+        // On value
+        ledValue = 0x7E;
+    }
```

Comment:
Wouldn't this suffice?
```javascript
// Custom value for multi-colored LEDs
var ledValue = value ? 0x7E : 0x7C;
```

Reply:
Ah ok, `value` can also have values other than `0`, `1`, `true` or `false` that are passed through without change, right?

Reply:
Right, some buttons on the controller provide multiple colors for the LED. In this case the no default value is used as mentioned in the comment.